deepchem/models/torch_models/layers.py

```diff
@@ -3058,6 +3058,135 @@ def forward(self, inputs: torch.Tensor):
         return atom_enbeddings
 
 
+class MolGANConvolutionLayer(nn.Module):
+    """
+    Graph convolution layer used in MolGAN model.
+    MolGAN is a WGAN type model for generation of small molecules.
+    Not used directly, higher level layers like MolGANMultiConvolutionLayer use it.
+    This layer performs basic convolution on one-hot encoded matrices containing
+    atom and bond information. This layer also accepts three inputs for the case
+    when convolution is performed more than once and results of previous convolution
+    need to used. It was done in such a way to avoid creating another layer that
+    accepts three inputs rather than two. The last input layer is so-called
+    hidden_layer and it hold results of the convolution while first two are unchanged
+    input tensors.
+
+    Example
+    -------
+    See: MolGANMultiConvolutionLayer for using in layers.
+
+    >>> import torch
+    >>> import torch.nn as nn
+    >>> import torch.nn.functional as F
+    >>> vertices = 9
+    >>> nodes = 5
+    >>> edges = 5
+    >>> units = 128
+
+    >>> layer1 = MolGANConvolutionLayer(units=units, edges=edges, nodes=nodes, name='layer1')
+    >>> adjacency_tensor = torch.randn((1, vertices, vertices, edges))  # added batch dimension
+    >>> node_tensor = torch.randn((1, vertices, nodes))  # added batch dimension
+    >>> output = layer1([adjacency_tensor, node_tensor])
+
+    References
+    ----------
+    .. [1] Nicola De Cao et al. "MolGAN: An implicit generative model
+        for small molecular graphs", https://arxiv.org/abs/1805.11973
+    """
+
+    def __init__(self,
+                 units,
```

Comment:
Make sure to add type annotations for all arguments

Reply:
done